### PR TITLE
Carousel: Visually indicate that an image has comments 

### PR DESCRIPTION
**Pull request by ramonjd**
Fixes https://github.com/Automattic/view-design/issues/312

#### Changes proposed in this Pull Request:

This PR adds a visual indicator plus comment count to communicate that a carousel image has comments.

**Dark theme**

Screen Shot 2021-06-29 at 11 47 06 am

**Light theme**

Screen Shot 2021-06-29 at 11 45 08 am

#### Jetpack product discussion

Motivation p58i-aLm-p2
Design p9Jlb4-2cI-p2

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
1. Create a tiled gallery, making sure the carousel is activated in `Jetpack Settings -> Writing`
2. Notice the comment icon on each gallery image slide
3. Comment on an image and refresh the page
4. You should see the comment indicator on the slides that have comments. Make sure the comment count corresponds with the number of comments.

### CODE REVIEW

**Comment by apeatling** [comment]
Could we instead use a comment count? I'm concerned that we use this kind of dot elsewhere when there is something new to look at. In this instance it's not new comments, just that it has comments.

**Review by apeatling — DISMISSED** [review]
See above comments

**Comment by olaolusoga** [comment]
Agree with using comment count. Dots are overdone, plus comment count provides context on the number of comments behind the click.

**Comment by ramonjd** [comment]
> Could we instead use a comment count? I'm concerned that we use this kind of dot elsewhere when there is something new to look at. In this instance it's not new comments, just that it has comments.

Cheers for feedback!

I did consider it, but baulked because we'd have to take formatting and localization into account and it would extend the JS. For example, `1,000` and `1.000`.

I acknowledge comments in the thousands are rare so if folks are happy with the raw integer for now, then I'll add it in.

**Comment by olaolusoga** [comment]
Let's go with the count for now, and improve when we run into that case (over 1000 comments).

File: projects/plugins/jetpack/modules/carousel/jetpack-carousel.css [hunk 1]
```diff
@@ -1058,6 +1058,27 @@ textarea#jp-carousel-comment-form-comment-field:focus::-webkit-input-placeholder
 	display: inline-block;
 }
 
+.jp-carousel-icon-comments .jp-carousel-has-comments-indicator {
+    display: none;
+    font-size: 12px;
```
Comment (andrewserong) [inline projects/plugins/jetpack/modules/carousel/jetpack-carousel.css:1063]: I noticed while testing this on a theme that sets a really light font weight (`300`) that its style bleeds into this element. Shall we add a `font-weight: normal` to this one, too?

image

(This screenshot is with Maywood theme active)

**Review by andrewserong — COMMENTED** [review]
This is looking great @ramonjd! I just noticed too tiny issues:

* I left a comment about font-weight, it looks like some themes' font-weight on `body` can bleed into the number indicator.

* When I open up the carousel on mobile, it takes a second or two to fetch the comments (naturally), but once they're loaded and the indicator appears, the presence of the indicator affects the horizontal position of the icon, so it bumps left a tiny bit. I was wondering if it'd help to make the comments indicator absolutely positioned so it doesn't affect the positions of the other icons?

Here's a screen grab:

https://user-images.githubusercontent.com/14988353/123726556-8b32e680-d8d3-11eb-9dab-40ef8a8a05fc.MP4

Looks really nice, though!

File: projects/plugins/jetpack/modules/carousel/jetpack-carousel.css (hunk 1 above)
Comment (ramonjd) [reply projects/plugins/jetpack/modules/carousel/jetpack-carousel.css:1063]: Thanks for testing @andrewserong ! Good tip, will do.

**Comment by ramonjd** [comment]
> I was wondering if it'd help to make the comments indicator absolutely positioned so it doesn't affect the positions of the other icons?

I also balked at this due to variable widths. 

Nevertheless, I'll play with left positions rather than fixing it to the right side. The icon's width should be static enough to support that. Cheers!

**Comment by andrewserong** [comment]
> Nevertheless, I'll play with left positions rather than fixing it to the right side. The icon's width should be static enough to support that.

I was actually just wondering if it'd work to set `.jp-carousel-icon` to `position: relative`, then move the `.jp-carousel-has-comments-indicator` up to be within that element (a sibling of `svg`), and set `position: absolute` on it, so that it can still visually be on the right side of the icon? You might have already tried something like that / this could be a different kind of fiddly, though, so I'll leave you to it 🙂

**Comment by andrewserong** [comment]
> I also balked at this due to variable widths.

Ah, I hadn't thought of that... ignore me!

**Comment by ramonjd** [comment]
> I was actually just wondering if it'd work to set .jp-carousel-icon to position: relative, then move the .jp-carousel-has-comments-indicator up to be within that element (a sibling of svg), and set position: absolute on it, so that it can still visually be on the right side of the icon? You might have already tried something like that 

I tried exactly that! :) Maybe I was being too pedantic, but I wanted the top, left value of the absolutely position element to never change. I'll try using left instead of right, right? :trollface:

**Review by andrewserong — DISMISSED** [review]
Excellent, thanks for fixing that up @ramonjd! This feels nice and solid in the UI now. Tested in Chrome, Safari, FF, and Edge on desktop, and Safari and Chrome on iOS mobile (iPhone 8). Also tested that nothing breaks if I switch off comments in http://localhost:4759/wp-admin/admin.php?page=jetpack#/writing 👍

**Review by apeatling — DISMISSED** [review]
Much better. 👍

**Review by jeherve — APPROVED** [review]
This tests well for me. Merging.

**Comment by jeherve** [comment]
Cherry-picked to `jetpack/branch-9.9` in bdd132c25765fd424349cedf43393ba5f24da9b5

**Pull request merged**